app/models/user.rb

```diff
@@ -1245,6 +1245,11 @@ def user_fields(field_ids = nil)
     end
   end
 
+  def public_user_field_values
+    @public_user_field_ids ||= UserField.where("show_on_profile OR show_on_user_card").pluck(:id)
```

Comment:
Nice one about public fields :slightly_smiling_face: 
I think it would be best to have this logic inside the `UserField` class though:
```ruby
  scope :public_fields, -> { where(show_on_profile: true).or(where(show_on_user_card: true)) }
```
Then it’s just a matter of doing
```ruby
    user_fields(UserField.public_fields.pluck(:id)).values.join(" ")
```